src/controllers/hid/hidcontroller.cpp

```diff
@@ -255,6 +255,35 @@ int HidController::close() {
     return 0;
 }
 
+QList<int> HidController::getInputReport(unsigned int reportID) {
+    Trace hidRead("HidController getInputReport");
+    unsigned char* pPreviousBuffer = m_pPollData[m_iPollingBufferIndex];
+    const int currentBufferIndex = (m_iPollingBufferIndex + 1) % kNumBuffers;
+    unsigned char* pCurrentBuffer = m_pPollData[currentBufferIndex];
+    int bytesRead;
+    pCurrentBuffer[0] = reportID;
+    bytesRead = hid_get_input_report(m_pHidDevice, pCurrentBuffer, kBufferSize);
+    controllerDebug(bytesRead << "bytes received by hid_get_input_report" << getName()
+                              << "serial #" << hid_serial
+                              << "(including report ID of" << reportID << ")");
+    bytesRead -= kReportIdSize;
```

Comment:
Stripping off the last byte unconditionally is wrong in many ways.

Depending on the arguments the report ID will be included in the result or not. And it will be included as the first byte, not the last byte. Please refer to the implementation in hid.c.

Reply:
The Windows implementation of hid_get_input_report in HIDAPI is wrong, it returns always a byte count incremented by 1. I started a upstream PR to fix this: https://github.com/libusb/hidapi/pull/232
I don't know, if the implementations for the other platforms return the correct byte count.
Nethertheless, this has no influence on the Mixxx application logic. You just see one byte more than read from the device.